circuit_passes/src/passes/loop_unroll/body_extractor.rs

```diff
@@ -282,109 +269,6 @@ impl LoopBodyExtractor {
         func_name
     }
 
-    fn new_custom_fn_load_bucket(
-        bucket: &dyn ObtainMeta,
-        load_fun: &str,
-        addr_type: AddressType,
-        location: InstructionPointer,
-    ) -> InstructionPointer {
-        LoadBucket {
-            id: new_id(),
-            source_file_id: bucket.get_source_file_id().clone(),
-            line: bucket.get_line(),
-            message_id: bucket.get_message_id(),
-            address_type: addr_type,
-            src: LocationRule::Indexed { location, template_header: None },
-            bounded_fn: Some(String::from(load_fun)),
-        }
-        .allocate()
-    }
```

Comment:
All this removed code and the one below I assume is because now that we have two passes that transform function bodies you needed to generalize a bit?

Reply:
Yes, they need to use the same helper functions.